lib/rules/prefer-template.js

```diff
@@ -192,32 +195,35 @@ module.exports = {
                 const plusSign = sourceCode.getFirstTokenBetween(currentNode.left, currentNode.right, token => token.value === "+");
                 const textBeforePlus = getTextBetween(currentNode.left, plusSign);
                 const textAfterPlus = getTextBetween(plusSign, currentNode.right);
+                const cleanTextAroundPlus = textBeforePlus.replace(/ +$/u, "") +
+                  textAfterPlus.replace(/^ +/u, "");
+
                 const leftEndsWithCurly = endsWithTemplateCurly(currentNode.left);
                 const rightStartsWithCurly = startsWithTemplateCurly(currentNode.right);
 
                 if (leftEndsWithCurly) {
 
                     // If the left side of the expression ends with a template curly, add the extra text to the end of the curly bracket.
                     // `foo${bar}` /* comment */ + 'baz' --> `foo${bar /* comment */  }${baz}`
-                    return getTemplateLiteral(currentNode.left, textBeforeNode, textBeforePlus + textAfterPlus).slice(0, -1) +
+                    return getTemplateLiteral(currentNode.left, textBeforeNode, cleanTextAroundPlus).slice(0, -1) +
                         getTemplateLiteral(currentNode.right, null, textAfterNode).slice(1);
                 }
                 if (rightStartsWithCurly) {
 
                     // Otherwise, if the right side of the expression starts with a template curly, add the text there.
                     // 'foo' /* comment */ + `${bar}baz` --> `foo${ /* comment */  bar}baz`
                     return getTemplateLiteral(currentNode.left, textBeforeNode, null).slice(0, -1) +
-                        getTemplateLiteral(currentNode.right, textBeforePlus + textAfterPlus, textAfterNode).slice(1);
+                        getTemplateLiteral(currentNode.right, cleanTextAroundPlus, textAfterNode).slice(1);
                 }
 
                 /*
                  * Otherwise, these nodes should not be combined into a template curly, since there is nowhere to put
                  * the text between them.
                  */
-                return `${getTemplateLiteral(currentNode.left, textBeforeNode, null)}${textBeforePlus}+${textAfterPlus}${getTemplateLiteral(currentNode.right, textAfterNode, null)}`;
+                return `${getTemplateLiteral(currentNode.left, cleanTextBeforeNode, null)}${textBeforePlus}+${textAfterPlus}${getTemplateLiteral(currentNode.right, cleanTextAfterNode, null)}`;
             }
 
-            return `\`\${${textBeforeNode || ""}${sourceCode.getText(currentNode)}${textAfterNode || ""}}\``;
+            return `\`\${${cleanTextBeforeNode}${sourceCode.getText(currentNode)}${cleanTextAfterNode}}\``;
```

Comment:
Would it work the same if we trim spaces only at this point?

```diff
-return `\`\${${textBeforeNode || ""}${sourceCode.getText(currentNode)}${textAfterNode || ""}}\``;
+return `\`\${${(textBeforeNode || "").replace(/^ +/u, "")}${sourceCode.getText(currentNode)}${(textAfterNode || "").replace(/ +$/u, "")}}\``;
```

Reply:
The `cleanTextAroundPlus` is required for cases with the `+` operator. For example:
```
var foo = 'bar' + baz;
// turns into
var foo = `bar${ baz}`;
// but it should be
var foo = `bar${baz}`;
```

But you're definitely right about line 223! I reverted this change